pkg/dependencies/main.go

```diff
@@ -43,16 +43,27 @@ func Init(ctx context.Context) *EdgeAPIServices {
 	}
 }
 
-type key int
+type servicesKeyType int
 
-// Key is the context key for dependencies on the request context
-const Key key = iota
+// servicesKey is the context key for dependencies on the request context
+const servicesKey servicesKeyType = iota
+
+// ContextWithServices add edge apis services to context
+func ContextWithServices(ctx context.Context, services *EdgeAPIServices) context.Context {
+	return context.WithValue(ctx, servicesKey, services)
+}
+
+// ServicesFromContext return the edge api services from context
+func ServicesFromContext(ctx context.Context) (*EdgeAPIServices, bool) {
+	edgeAPIServices, ok := ctx.Value(servicesKey).(*EdgeAPIServices)
+	return edgeAPIServices, ok
+}
```

Comment:
We should catch any errors as we can, I have 2 suggestions:
1. Return only `*EdgeAPIServices` and ignore the check of its existence, then on each `services, _ := dependencies.ServicesFromContext(r.Context())` we can write `services := dependencies.ServicesFromContext(r.Context())` & check `services` for nil.
  ```suggestion
  func ServicesFromContext(ctx context.Context) *EdgeAPIServices {
	      edgeAPIServices, _ := ctx.Value(servicesKey).(*EdgeAPIServices)
	      return edgeAPIServices
  }
  ```
2. Keep this function as it is & for each `services, _ := dependencies.ServicesFromContext(r.Context())` we should write `services, ok := dependencies.ServicesFromContext(r.Context())`, then try to see if `!ok`

I tend to go with the first suggestion, but this is up to you.

Reply:
if context value does not exist a nil will be returned , and the cast will fail with ok = false , so the check will be from the side of the caller function.

Reply:
Fixing caller function is not the purpose of this PR.

Reply:
@Avielyo10 what do you think?

Reply:
You are correct saying this is not the PR intention, but if we do a refactor don't you think we should make it all through it is done?so this won't need more refactoring on the same place?
You can create a separate PR for the check if you'd like, this will be very helpful for everyone

You are also correct that its the caller responsibility to check the output of a function, but then I'd suggest sending back an `error` instead of a boolean, this way static code analysis & security checks we want to add will fail since we have an unhandled error. (returning `bool` won't do that)

Reply:
I think we should not modify ServicesFromContext to return an error , the caller function should generate it's proper error when receiving ok = false , perhaps the caller function will call init function and write a new value , some other function will generate a fatal error, and other function can pass silent if it's not critical, we cannot decide at this place.

Reply:
Maybe there is a sync-up misunderstanding :) let me explain my point of view

When accessing Context values the code reader is waiting to see a boolean as in all the project we are using that pattern when accessing context values.

Returning errors is a good a idea , as we the caller function will have a ready to use error to return when necessary , but the error should be an instance so the caller can make comparisons.

But we not want to make services context value a special case.

I think If we should return an error let's do it by writing a global public function that access context and return standard project errors like 
ErtContextVaslueNotFound  - in case of key not found
ErrContextValueMismatch  - In case the cast fail

Let's discuss about it after the holidays, Happy Holidays :)

Reply:
@ldjebran I think you are missing Golang's error concept, returning an error doesn't mean throwing an error. Errors are values, so each caller can handle it differently but as it comes to code quality check then the caller must handle it & can't ignore it.
```suggestion
func ServicesFromContext(ctx context.Context) (edgeAPIServices *EdgeAPIServices, err error) {
	edgeAPIServices, ok = ctx.Value(servicesKey).(*EdgeAPIServices)
	if !ok {
	    err = errors.New("Couldn't get services from Context")
	}
}
```

Reply:
I think we can catch the error here and log as log.Fatal. If something happens, the caller shouldn't have to handle it because this is core functionality.

Reply:
@biancarosa I was thinking maybe if there is an error there, then we can return immediately on `Middleware` without forwarding.